bigquery_etl/cli/query.py

```diff
@@ -506,76 +507,89 @@ def _backfill_query(
     run_checks,
 ):
     """Run a query backfill for a specific date."""
+    backfill_date_str = backfill_date.strftime("%Y-%m-%d")
+    if backfill_date_str in exclude:
+        click.echo(f"Skip {query_file_path} backfill for run date {backfill_date_str}")
+        return True
+
     project, dataset, table = extract_from_query_path(query_file_path)
 
     match partitioning_type:
         case PartitionType.DAY:
-            partition = backfill_date.strftime("%Y%m%d")
+            partition_date = backfill_date + timedelta(days=date_partition_offset)
+            partition = partition_date.strftime("%Y%m%d")
         case PartitionType.MONTH:
+            if date_partition_parameter != 0:
+                # TODO: Support offsets here e.g. desktop_mobile_search_clients_monthly_v1
+                raise ValueError("Unsupported offset for month partitions.")
             partition = backfill_date.strftime("%Y%m")
         case _:
             raise ValueError(f"Unsupported partitioning type: {partitioning_type}")
 
-    backfill_date = backfill_date.strftime("%Y-%m-%d")
-    if backfill_date not in exclude:
-        if destination_table is None:
-            destination_table = f"{project}.{dataset}.{table}"
+    if destination_table is None:
+        destination_table = f"{project}.{dataset}.{table}"
 
-        if not no_partition:
-            destination_table = f"{destination_table}${partition}"
+    if not no_partition:
+        destination_table = f"{destination_table}${partition}"
 
-        if not QUALIFIED_TABLE_NAME_RE.match(destination_table):
-            click.echo(
-                "Destination table must be named like:" + " <project>.<dataset>.<table>"
-            )
-            sys.exit(1)
+    if not QUALIFIED_TABLE_NAME_RE.match(destination_table):
+        click.echo("Destination table must be named like: <project>.<dataset>.<table>")
+        sys.exit(1)
 
-        click.echo(
-            f"Run backfill for {destination_table} "
-            f"with @{date_partition_parameter}={backfill_date}"
-        )
+    scheduling_params = []
+    if date_partition_parameter is not None:
+        # TODO: Support non date_partition_parameter scheduling.parameters e.g. clients_first_seen_28_days_later_v1
```

Comment:
Is this possible? These tables would have failed above with no partitioning type, right?

Reply:
This matches the Airflow logic, `date_partition_offset` even with a null `date_partition_parameter` adds a partition template. `date_partition_parameter` can be null (but the table still partitioned) if you want to run your query with something other than the partition field as a "daily" query parameter. In the TODO example the table is partitioned by `first_seen_date` but the query is run with today's `submission_date`.

This is kinda confusing but `date_partition_parameter` exists as a convenient default for tables that are partitioned by the same field they're meant to be updated daily by

Reply:
I think what is confusing me _is_ the TODO. This currently does not support that situation? But we want to support it eventually? Then the comment "These tables would have failed above with no partitioning type, right?" is still correct, but we will eventually change that behavior?

Reply:
Yes this isn't currently supported. Some queries set additional scheduling `parameters` using Airflow syntax (e.g. `submission_date:DATE:{{ds}}`) in metadata, those still aren't supported. This PR only adds support for `date_partition_offset` The goal here is that you should be able to backfill any table with a query that's run in Airflow